frontend/src/components/editor/DescriptionEditor.js

```diff
@@ -3,40 +3,173 @@ import PropTypes from 'prop-types';
-        onValidate={() => onValidate('spec.description')}
-      />
-      <MarkdownEditor
-        title="Details about the Operator"
-        description="Please describe the features your Operator provides to manage the application and what is it's value add."
-        markdown={operator.aboutMarkdown}
-        onChange={markdown => onUpdate(markdown, 'aboutMarkdown')}
-        onValidate={() => onValidate('aboutMarkdown')}
-      />
-      <MarkdownEditor
-        title="Details the prerequisites for enabling the Operator"
-        description="Please describe any steps a user needs to take prior to enabling this Operator (e.g. any Secrets or ConfigMaps that need to be in place upfront)."
-        markdown={operator.prerequisitesMarkdown}
-        onChange={markdown => onUpdate(markdown, 'prerequisitesMarkdown')}
-        onValidate={() => onValidate('prerequisitesMarkdown')}
-      />
-    </div>
-  </React.Fragment>
-);
+class DescriptionEditor extends React.Component {
+  state = {
+    application: '',
+    about: '',
+    prerequisites: '',
+    applicationError: null,
+    aboutError: null,
+    prerequisitesError: null
+  };
+
+  static getDerivedStateFromProps(props) {
+    const markdown = DescriptionEditor.splitDescriptionIntoSections(props.operator);
+    const errors = {
+      applicationError: DescriptionEditor.validateMarkdown(markdown.application),
+      aboutError: DescriptionEditor.validateMarkdown(markdown.about),
+      prerequisitesError: DescriptionEditor.validateMarkdown(markdown.prerequisites)
+    };
+
+    return {
+      ...markdown,
+      ...errors
+    };
+  }
+
+  static splitDescriptionIntoSections(operator) {
+    const description = _.get(operator, 'spec.description', '');
+    let application = description;
+    let about = '';
+    let prerequisites = '';
+
+    const aboutHeaderIndex = description.indexOf(OPERATOR_DESCRIPTION_ABOUT_HEADER);
+    const prerequisitesHeaderIndex = description.indexOf(OPERATOR_DESCRIPTION_PREREQUISITES_HEADER);
+
+    // if we can identify headers, split using them
+    // at least one header must be available
+    if (aboutHeaderIndex > -1) {
+      if (prerequisitesHeaderIndex > -1) {
+        about = description.substring(aboutHeaderIndex, prerequisitesHeaderIndex);
+        prerequisites = description.substring(prerequisitesHeaderIndex);
+      } else {
+        about = description.substring(aboutHeaderIndex);
+      }
+
+      application = description.substring(0, aboutHeaderIndex);
+    } else if (prerequisitesHeaderIndex > -1) {
+      application = description.substring(0, prerequisitesHeaderIndex);
+      prerequisites = description.substring(prerequisitesHeaderIndex);
+
+      // no our headers found, trying to match using level 2 headers
+    } else {
+      // contains splitted sections and headers using capture group inbetween splitted sections
+      const segments = description.split(/(## [^\r?\n]+)/);
```

Comment:
This splits up the H3's (###) that exist in markdown loaded from CSVs into an H2 and an H1:

![image](https://user-images.githubusercontent.com/11633780/57015122-1ecbd800-6be1-11e9-9371-51e198baf7a6.png)


I believe @dmesser said that if we don't find our defined headers, put all the markdown in the first section. I think putting random pieces into the differently titled sections would be misleading.

Reply:
Good catch, somehow did not add line start.
I got it that we should use h2 as well otherwise its useless for import.

Reply:
Talked to Daniel and we should use ## as well and pick the first 3 we find if our specific headers are not present.